recursive-feature-elimination/script.whizzml

```diff
@@ -0,0 +1,248 @@
+                    selected (keys (dissoc imps removed-field))
+                    _ (log-info "There are " 
+                                (count selected) 
+                                " fields after filtering")
+                    ev (when do-evaluation 
+                           (fetch(evaluate-model rf-id test-dataset-id)))
+                    ev-result (when ev (ev ["result" "model" evaluation-metric]))
+                    ;; Iterations evaluation info
+                    iterations (when ev {"features" (count selected)
+					 "evaluation"  ev-result
+					 "last-removed" removed-fields-names})
+                    evaluations (when ev (append evaluations iterations)))
+              (recur selected
+                      evaluations))))))
+
+;; response
+;;
+;; Creates a human-readable response and cleans the created resources
+;;
+;; Inputs:
+;;   selected: (list) Selected fields ids
+;;   selected-names: (list) Selected fields names
+;;   evaluations: (list) Result of the evaluations at each iteration
+;;   do-evaluation: (boolean) Whether we performed evaluations
+;;
+;; Output: (map) Map with selected field names, selected field ids, and 
+;;         iterations evaluation info (only if test-dataset-id is present)
+(define (response selected selected-names evaluations do-evaluation)
+  ;; Remove created resources
+  (map safe-delete (created-resources))
+  (let( evaluations (if  do-evaluation
+                         evaluations
+                         ["test-dataset-id needed to create evaluations"]))
+    {"selected-fields" selected
+     "selected-fields-names" selected-names
+     "evaluations" evaluations}))
+
+
+;; feature-names
+;;
+;; Returns feature names given ids
+;;
+;; Inputs:
+;;   fieds: (map) Fields map (Output from resource-fields)
+;;   ids: (list) List of field ids
+;;
+;; Output: (list) List with all the field names
+;;
+(define (feature-names fields ids)
+    (map (lambda (id) (fields [id "name"])) ids))
+
+
+;; get-objective-name
+;;
+;; Returns the name of the field used as objective field
+;;
+;; Inputs:
+;;  dataset: (map) Dataset resource info
+;;  objective-id: (string) id of the objective field
+;;
+;; Outputs: (string) Name of the objective field
+;;
+(define (get-objective-name dataset objective-id)
+  (dataset ["fields" objective-id "name"] false))
+
+;; get-objective
+;;
+;; Retrieves and checks whether the objective field of the given
+;; dataset exists, raising an error otherwise.
+;;
+;; Inputs:
+;;  ds-id: (dataset-id) Dataset id
+;;  obj-id: (string) id of the objective field
+;;
+;; Outputs: (string) Same objective field id
+;;
+(define (get-objective ds-id obj-id)
+  (let (obj-id (if (empty? obj-id)
+                   (dataset-get-objective-id ds-id)
+                   obj-id)
+        fields (resource-fields ds-id)
+        obj-id ((find-field fields obj-id) "id")
+        otype (fields [obj-id "optype"] "missing"))
+    (when (= "missing" otype)
```

Comment:
no big deal, but it's a bit more idiomatic to use `false` as the not-found value, and then:

```
(when (not otype)
  (raise ...))
```